projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts

```diff
@@ -401,6 +406,16 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
     public set batchEditing(_val: boolean) {
     }
 
+    public get selectedRows(): any[] {
+        const selectedRowKeys = this.selectionService.getSelectedRows();
+        const selectedRowIds = [];
+        this.selectedPivotKeys.forEach((value, key) => {
```

Comment:
Since this collection is populated in a getter in the rows, if a selected record has not entered the data view yet (i.e. you need to scroll down to load a row for it) it will not be part of the selected rows. It may be better to check the full data collection (the result of all the pipes) for selection id matches and return the related records.